bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -190,7 +190,7 @@ func (r *KubeadmConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	// Lookup the cluster the config owner is associated with
 	cluster, err := util.GetClusterByName(ctx, r.Client, configOwner.GetNamespace(), configOwner.ClusterName())
 	if err != nil {
-		if errors.Cause(err) == util.ErrNoCluster {
+		if errors.Is(errors.Cause(err), util.ErrNoCluster) {
 			log.Info(fmt.Sprintf("%s does not belong to a cluster yet, waiting until it's part of a cluster", configOwner.GetKind()))
 			return ctrl.Result{}, nil
 		}
```

Comment:
```suggestion
```
Unreachable code. THis error is never returned by GetClusterByName

Reply:
Surely, this error is returned by only GetClusterFromMetadata. Thanks